app/src/main/java/de/danoeh/antennapod/fragment/FeedInfoFragment.java

```diff
@@ -211,25 +226,50 @@ private void showFeed() {
 
         txtvDescription.setText(description);
 
-        if (!TextUtils.isEmpty(feed.getAuthor())) {
-            txtvAuthor.setText(feed.getAuthor());
-            txtvAuthorHeader.setText(feed.getAuthor());
-        } else {
-            lblAuthor.setVisibility(View.GONE);
-            txtvAuthor.setVisibility(View.GONE);
-        }
-        if (!TextUtils.isEmpty(feed.getLanguage())) {
-            txtvLanguage.setText(LangUtils.getLanguageString(feed.getLanguage()));
-        } else {
-            lblLanguage.setVisibility(View.GONE);
-            txtvLanguage.setVisibility(View.GONE);
-        }
         txtvUrl.setText(feed.getDownload_url() + " {fa-paperclip}");
         Iconify.addIcons(txtvUrl);
-
         getActivity().invalidateOptionsMenu();
     }
 
+    private void loadStatistics() {
+        if (disposable != null) {
+            disposable.dispose();
+        }
+
+        disposable =
+                Observable.fromCallable(() -> {
+                    List<StatisticsItem> statisticsData = DBReader.getStatistics();
+
+                    for (StatisticsItem statisticsItem : statisticsData) {
+                        if (statisticsItem.feed.getId() == feed.getId()) {
+                            podcastStatistics = statisticsItem;
+                            break;
+                        }
+                    }
+
+                    return podcastStatistics;
+                })
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(result -> {
+                            if (podcastStatistics != null) {
+                                txtvPodcastTime.setText(Converter.shortLocalizedDuration(
+                                        getContext(), podcastStatistics.timePlayed));
+                                txtvPodcastSpace.setText(Formatter.formatShortFileSize(
+                                        getContext(), podcastStatistics.totalDownloadSize));
+                                txtvPodcastEpisodeCount.setText(String.format(Locale.getDefault(),
+                                        "%d%s", podcastStatistics.episodesDownloadCount,
+                                        getString(R.string.episodes_suffix)));
+                            } else {
```

Comment:
Please move the `else` part to the `error ->` section below. It is then called in both cases, when the loading code above fails or when it returns null.

Reply:
You no longer need the `else` part here in the success section. RxJava automatically calls the `error ->` section when the result is `null`.